src/CommonITILValidation.php

```diff
@@ -770,8 +804,34 @@ public static function getNumberToValidate($users_id)
     final public static function getTargetCriteriaForUser(int $users_id, bool $search_in_groups = true): array
     {
         $target_criteria = [
-            static::getTableField('itemtype_target') => User::class,
-            static::getTableField('items_id_target') => $users_id,
+            'OR' => [
+                [
+                    static::getTableField('itemtype_target') => User::class,
+                    static::getTableField('items_id_target') => $users_id,
+                ],
+                'AND' => [
+                    [
+                        'OR' => [
+                            [
+                                User::getTableField('substitution_start_date')  => null,
+                            ],
+                            [
+                                User::getTableField('substitution_start_date')  => ['<=', $_SESSION['glpi_currenttime']],
+                            ],
+                        ],
+                    ], [
+                        'OR' => [
+                            [
+                                User::getTableField('substitution_end_date') => null,
+                            ],
+                            [
+                                User::getTableField('substitution_end_date') => ['>=', $_SESSION['glpi_currenttime']],
```

Comment:
You should use `new \QueryExpression('NOW()')`, in order to ensure that date will use the timezone defined in MySQL session.
```suggestion
                                User::getTableField('substitution_end_date') => ['>=', new \QueryExpression('NOW()')],
```

Reply:
solved with https://github.com/glpi-project/glpi/commit/ac6de93d6ad23a2b46295229e50cd11dc377fe4b